File: Part1_강의자료/Ch05_정렬/문제별코드/10814_나이순정렬/10814.cpp
#include <stdio.h>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

struct Member {
    int age;
    string name;
    int idx;
    bool operator<(const Member &o) const{
        if (age != o.age)
            return age < o.age;
        return idx < o.idx;
    }
};

int main()
{
    int N;
    vector<Member> members;
    int age;
    char name[101];

    scanf("%d", &N);
    for (int i = 0; i < N; i++)
    {
        scanf("%d %s", &age, name);
        members.push_back({age, name, i});
    }

    sort(members.begin(), members.end());

    for (int i = 0; i < N; i++)
        printf("%d %s\n", members[i].age, members[i].name.c_str());

    return 0;
}

File: Part1_강의자료/Ch05_정렬/문제별코드/10814_나이순정렬/10814.py
class Member:
    def __init__(self, age: int, name: str, idx: int):
        self.age = age
        self.name = name
        self.idx = idx

N = int(input())
members = []
for i in range(N):
    age, name = input().split()
    members.append(Member(int(age), name, i))

members.sort(key=lambda x: (x.age, x.idx))

for member in members:
    print(member.age, member.name)
